python_modules/dagster/dagster/core/execution/plan/state.py

```diff
@@ -49,17 +50,26 @@ class KnownExecutionState(
     resolved dynamic outputs.
     """
 
-    def __new__(cls, previous_retry_attempts, dynamic_mappings, step_output_versions=None):
+    def __new__(
+        cls,
+        previous_retry_attempts: Optional[Dict[str, int]] = None,
+        dynamic_mappings: Optional[Dict[str, Dict[str, List[str]]]] = None,
+        step_output_versions: Optional[List[StepOutputVersionData]] = None,
+        ready_outputs: Optional[Set[StepOutputHandle]] = None,
```

Comment:
Do we need to populate `ready_outputs` for the `for_reexecution` case?

Reply:
the `parent` stuff is next in the stack, this is just for the current run so should always start blank